File: src/main/java/com/topably/assets/trades/domain/dto/add/AddMoneyTradeDto.java
package com.topably.assets.trades.domain.dto.add;

import com.topably.assets.trades.domain.TradeOperation;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Setter
@Getter
public class AddMoneyTradeDto {

    private LocalDateTime date;
    private TradeOperation operation;
    private BigDecimal amount;
    private String currencyCode;
}

File: client/src/app/features/trades/types/money/add-money-trade.dto.ts
import { MoneyTradeOperation } from './money-trade-operation';
import { TradeCategory } from '../trade-category';
import { AddTradeDto } from '../add-trade.dto';

export interface AddMoneyTradeDto extends AddTradeDto {
  operation: MoneyTradeOperation;
  amount: number;
  currencyCode: string;
}
